mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -118,9 +118,11 @@ private[spark] object RandomForest extends Logging {
     val splits = findSplits(retaggedInput, metadata, seed)
     timer.stop("findSplits")
     logDebug("numBins: feature: number of bins")
-    logDebug(Range(0, metadata.numFeatures).map { featureIndex =>
-      s"\t$featureIndex\t${metadata.numBins(featureIndex)}"
-    }.mkString("\n"))
+    if (log.isDebugEnabled) {
+      logDebug(Range(0, metadata.numFeatures).map { featureIndex =>
```

Comment:
Also trivial, but I woulda written this as just `(0 until metadata.numFeatures.map(i => s"\t$i\t${metadata.numBins(i)}").mkString("\n")` I know it was like this already. Up to your taste.

Does this actually help much though? `logDebug` is already declared to take `(msg: => String)` meaning it's only evaluated lazily.